lib/pure/os.nim

```diff
@@ -1686,6 +1686,26 @@ proc setFilePermissions*(filename: string, permissions: set[FilePermission]) {.
       var res2 = setFileAttributesA(filename, res)
     if res2 == - 1'i32: raiseOSError(osLastError(), $(filename, permissions))
 
+func toFilePermissions*(perm: Natural): set[FilePermission] {.since: (1, 3).} =
+  ## Convenience func to convert Unix like file permission to ``set[FilePermission]``.
+  ##
+  ## See also:
+  ## * `getFilePermissions <#getFilePermissions,string>`_
+  ## * `setFilePermissions <#setFilePermissions,string,set[FilePermission]>`_
+  runnableExamples:
+    doAssert toFilePerm(0o700) == {fpUserExec, fpUserRead, fpUserWrite}
+    doAssert toFilePerm(0o070) == {fpGroupExec, fpGroupRead, fpGroupWrite}
+    doAssert toFilePerm(0o007) == {fpOthersExec, fpOthersRead, fpOthersWrite}
+    doAssert toFilePerm(0o644) == {fpUserWrite, fpUserRead, fpGroupRead, fpOthersRead}
+    doAssert toFilePerm(0o777) == {fpUserExec, fpUserWrite, fpUserRead, fpGroupExec, fpGroupWrite, fpGroupRead, fpOthersExec, fpOthersWrite, fpOthersRead}
+    doAssert toFilePerm(0o000) == {}
+  var perm = uint perm
```

Comment:
```suggestion
func toFilePermissions*(perm: uint): set[FilePermission] {.since: (1, 3).} =
  ## Convenience func to convert Unix like file permission to ``set[FilePermission]``.
  ##
  ## See also:
  ## * `getFilePermissions <#getFilePermissions,string>`_
  ## * `setFilePermissions <#setFilePermissions,string,set[FilePermission]>`_
  runnableExamples:
    doAssert toFilePerm(0o700) == {fpUserExec, fpUserRead, fpUserWrite}
    doAssert toFilePerm(0o070) == {fpGroupExec, fpGroupRead, fpGroupWrite}
    doAssert toFilePerm(0o007) == {fpOthersExec, fpOthersRead, fpOthersWrite}
    doAssert toFilePerm(0o644) == {fpUserWrite, fpUserRead, fpGroupRead, fpOthersRead}
    doAssert toFilePerm(0o777) == {fpUserExec, fpUserWrite, fpUserRead, fpGroupExec, fpGroupWrite, fpGroupRead, fpOthersExec, fpOthersWrite, fpOthersRead}
    doAssert toFilePerm(0o000) == {}
  var perm = perm
```

Reply:
I'd generally prefer that we don't use `uint` sparingly. `Natural` is pretty much preferred over `uint` in Nim where possible.